packages/terra-date-picker/src/DatePicker.jsx

```diff
@@ -206,7 +206,7 @@ class DatePicker extends React.Component {
       const nextSelectedDate = DateUtil.createSafeDate(nextDateValue, initialTimeZone);
 
       return {
-        selectedDate: nextSelectedDate,
+        selectedDate: nextSelectedDate || prevState.selectedDate,
```

Comment:
this would still allow [selectedDate](https://github.com/cerner/terra-framework/blob/5e057423117f6c56f5d1767f7a45ee8e6a2c12ed/packages/terra-date-picker/src/DatePicker.jsx#L456) in picker to be cleared when both nextdate is invalid. 

clearing [selectedDateInPicker](https://github.com/cerner/terra-framework/blob/5e057423117f6c56f5d1767f7a45ee8e6a2c12ed/packages/terra-date-picker/src/DatePicker.jsx#L452) only when value is undefined or null would fix the issue shown on below gif 


```js
    if (this.isDefaultDateAcceptable && !this.isDateWithinRange(selectedDateInPicker)) {
      selectedDateInPicker = undefined;
    }
```
remove above condition and include it on [line no. 452](https://github.com/cerner/terra-framework/blob/5e057423117f6c56f5d1767f7a45ee8e6a2c12ed/packages/terra-date-picker/src/DatePicker.jsx#L452)
```
selectedDateInPicker = (this.isDefaultDateAcceptable && !this.isDateWithinRange(selectedDateInPicker)) ? undefined : this.state.selectedDate;
```

https://user-images.githubusercontent.com/12869809/205621793-f4e6d4b7-02d5-4d30-9542-d8454a50aa38.mov